kobo/client/__init__.py

```diff
@@ -183,6 +183,43 @@ def __getattr__(self, name):
         except:
             raise AttributeError("'%s' object has no attribute '%s'" % (self.__class__.__name__, name))
 
+    @property
+    def is_logged_in(self):
+        return self._logged_in
+
+    def login(self):
+        """Retry login, and return on success.
+
+        If Hub is already logged in, this method does not re-attempt it.
+
+        This class attempts login once in its constructor, but does not
+        raise on failure in that case. Callers can check `is_logged_in`
+        to determine if that attempt was successful.
+
+        Returns:
+            bool: Returns True if login was successful, or raises an exception
+                on failure. See Raises for more information.
+
+        Raises:
+            AuthenticationError when login fails.
+
+        """
+        if self.is_logged_in:
+            return True
+
+        self._login(force=True)
+
+        if not self.is_logged_in:
+            msg = (
+                "Login attempt failed. See logs for more information. "
+                "If the logs don't provide any indication as to what went wrong, "
+                "try passing HubProxy a logger at the `logging.DEBUG` level "
+                "for more verbose messages."
+            )
+            raise AuthenticationError(msg)
+
+        return True
```

Comment:
Why return True?

I'd rather not since it's misleading.  If someone sees that the return value is True then they might think they're supposed to write code such as:

```
if not hub.login():
  do_something_when_login_fails()
```

But the branch would be dead code.  I think it just creates confusion with respect to how the method should be used.

Reply:
Good point, I will make this change.